events/app.py

```diff
@@ -0,0 +1,119 @@
+import os
+import logging
+import datetime
+import pymongo
+# import pprint
+
+from flask import Flask, request, Response
+
+app = Flask(__name__)
+
+
+def db_setup():
+    """Connect to MongoDB Atlas database, returns events collection."""
+    MONGODB_ATLAS_USERNAME = os.environ.get(
+        "MONGODB_ATLAS_USERNAME")
+    MONGODB_ATLAS_PASSWORD = os.environ.get(
+        "MONGODB_ATLAS_PASSWORD")
+    MONGODB_ATLAS_CLUSTER_ADDRESS = os.environ.get(
+        "MONGODB_ATLAS_CLUSTER_ADDRESS")
+    PYMONGO_URI = "mongodb+srv://{}:{}@{}".format(
+        MONGODB_ATLAS_USERNAME,
+        MONGODB_ATLAS_PASSWORD,
+        MONGODB_ATLAS_CLUSTER_ADDRESS)
+    # print("Pymongo URI: {}".format(PYMONGO_URI))
+    client = pymongo.MongoClient(PYMONGO_URI)
+    eventsDB = client.eventsDB
+    return eventsDB.all_events
+
+
+@app.route('/v1/add', methods=['POST'])
+def add_event():
+    # TODO(cmei4444): Get event information from request data
+    info = request.form['info']
+    # Add information into database
+    current_time = datetime.datetime.now()
+    build_event_info(info, current_time)
+    event = Event(info)
+    if not event.is_valid():
```

Comment:
Why not have the Event constructor throw an exception if the supplied info is not valid?

Reply:
Constructing the namedtuple causes this exception now